core/src/main/clojure/xtdb/buffer_pool.clj

```diff
@@ -180,7 +180,9 @@
           (close [_]
             (util/close unl)
             (when (.isOpen file-ch)
-              (.close file-ch)))))))
+              (.close file-ch))
+            ;; If tmp path still present/hasn't been atomic moved, delete it on close
+            (Files/deleteIfExists tmp-path))))))
```

Comment:
Do we always call close when the `end` call errors (or `writeBatch`)? If that holds then that seems reasonable.

Reply:
Your tests suggest so, so all good.